engine/security/src/test/scala/pl/touk/nussknacker/ui/security/oauth2/JwtTokenAuthenticationSpec.scala

```diff
@@ -0,0 +1,85 @@
+package pl.touk.nussknacker.ui.security.oauth2
+
+import akka.http.javadsl.model.headers.HttpCredentials
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.{Directives, Route}
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import com.typesafe.config.ConfigFactory
+import de.heikoseeberger.akkahttpcirce.FailFastCirceSupport
+import org.scalatest.{FunSpec, Matchers}
+import pdi.jwt.{JwtAlgorithm, JwtCirce, JwtClaim}
+import pl.touk.nussknacker.ui.security.api.AuthenticationResources
+import pl.touk.nussknacker.ui.security.http.RecordingSttpBackend
+import sttp.client.testing.SttpBackendStub
+import sttp.model.Uri
+
+import java.net.URI
+import java.security.KeyPairGenerator
+import java.time.Clock
+import java.util.Base64
+
+class JwtTokenAuthenticationSpec extends FunSpec with Matchers with ScalatestRouteTest with Directives with FailFastCirceSupport {
+
+  implicit val clock: Clock = Clock.systemUTC()
+
+  private val keyPair = KeyPairGenerator.getInstance("RSA").generateKeyPair()
+
+  private val config = ConfigFactory.parseString(
+    s"""
+        authentication: {
```

Comment:
Hmm. shouldn't be added pipes at the beginning of each line to make `stripMargin` works? Intellij Idea always add it when u use text blocks. Take a look at: https://www.oreilly.com/library/view/scala-cookbook/9781449340292/ch01s03.html